test/e2e/apps/controller_revision.go

```diff
@@ -136,8 +136,8 @@ var _ = SIGDescribe("ControllerRevision", framework.WithSerial(), func() {
 		ginkgo.By("Check that daemon pods launch on every node of the cluster.")
 		err = wait.PollUntilContextTimeout(ctx, dsRetryPeriod, dsRetryTimeout, true, checkRunningOnAllNodes(f, testDaemonset))
 		framework.ExpectNoError(err, "error waiting for daemon pod to start")
-		err = e2edaemonset.CheckDaemonStatus(ctx, f, dsName)
-		framework.ExpectNoError(err)
+		err = wait.PollUntilContextTimeout(ctx, dsRetryPeriod, dsRetryTimeout, true, e2edaemonset.CheckDaemonStatus(ctx, f, dsName))
+		framework.ExpectNoError(err, "error waiting for daemonset to report all pods are scheduled and ready")
```

Comment:
I missed that you changed the prototype of a function in the e2e/framework so that it works well with `wait.PollUntilContextTimeout`. The revised function now also pollutes the test output with a log message on each retry.

I care less about tests (this code here), but I care about the e2e/framework enough that I'll submit a followup-PR... stay tuned.

> In order to avoid more churn code, change the CheckDaemonStatus signature to the wait.Condition type and use it in a async poll loop on the tests.

Even less code churn would have been to do the polling inside the helper function. Then tests wouldn't have to be modified at all.

Reply:
The reason why I noticed and why I care is because I am using `CheckDaemonStatus` in one of my branches for DRA where I am testing with a daemon set and that broke when rebasing.

Reply:
=> https://github.com/kubernetes/kubernetes/pull/128079

Reply:
Thanks